instrumentation/google.golang.org/grpc/otelgrpc/stats_handler_test.go

```diff
@@ -0,0 +1,40 @@
+package otelgrpc
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"google.golang.org/grpc/stats"
+)
+
+func TestHandleRPC(t *testing.T) {
+	const iteration = 100
+	wg := &sync.WaitGroup{}
+	ctx := context.Background()
+	h := NewClientHandler()
+	ctx = h.TagRPC(ctx, &stats.RPCTagInfo{
+		FullMethodName: "test/method",
+	})
+	for i := 0; i < iteration; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			h.HandleRPC(ctx, &stats.Begin{})
+			wg.Add(3)
+			go func() {
+				defer wg.Done()
+				h.HandleRPC(ctx, &stats.InPayload{})
+			}()
+			go func() {
+				defer wg.Done()
+				h.HandleRPC(ctx, &stats.OutPayload{})
+			}()
+			go func() {
+				defer wg.Done()
+				h.HandleRPC(ctx, &stats.End{})
+			}()
+		}()
+	}
+	wg.Wait()
+}
```

Comment:
I would reproduce a normal use. If https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4537 get merged this could probably be (in `test` subdirectory):

```go
func TestStatsHandlerConcurrentSafe(t *testing.T) {
	listener, err := net.Listen("tcp", "127.0.0.1:0")
	require.NoError(t, err, "failed to open port")
	client := newGrpcTest(t, listener,
		[]grpc.DialOption{ grpc.WithStatsHandler(otelgrpc.NewClientHandler() },
		[]grpc.ServerOption{ grpc.StatsHandler(otelgrpc.NewServerHandler() },
	)

	var wg sync.WaitGroup
	const n = 100
	for i := 0; i < n; i++ {
		wg.Add(1)
		go func() {
			defer wg.Done()
			doCalls(client)
		}()
	}
	wg.Wait()
}
```

Reply:
Nice thanks for the pointer! I wasn't sure how to properly simulate gRPC calls in tests. Will move this

Reply:
https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4537 is merged so you can reuse the refactored `newGrpcTest` helper